signer/src/schnorr.rs

```diff
@@ -176,7 +176,7 @@ impl<H: 'static + Hashable> Schnorr<H> {
             .to_roinput()
             .append_field(kp.public.point().x)
             .append_field(kp.public.point().y)
-            .append_scalar(kp.secret.scalar().clone())
+            .append_scalar(*kp.secret.scalar())
```

Comment:
Can you squash this with the commit actually introducing this change, please? We want to avoid this kind of split commits to ease reverts if necessary.